controllers/clusters/postgresql_controller.go

```diff
@@ -468,6 +482,129 @@ func (r *PostgreSQLReconciler) handleUpdateCluster(
 	return models.ExitReconcile
 }
 
+func (r *PostgreSQLReconciler) createUser(
+	ctx context.Context,
+	l logr.Logger,
+	c *v1beta1.PostgreSQL,
+	uRef *v1beta1.UserReference,
+) error {
+	req := types.NamespacedName{
+		Namespace: uRef.Namespace,
+		Name:      uRef.Name,
+	}
+
+	u := &clusterresourcesv1beta1.PostgreSQLUser{}
+	err := r.Get(ctx, req, u)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			l.Error(err, "Cannot create a PostgreSQL user. The resource is not found", "request", req)
+			r.EventRecorder.Eventf(c, models.Warning, models.NotFound,
+				"User is not found, create a new one PostgreSQL User or provide correct userRef."+
+					"Current provided reference: %v", uRef)
+			return err
+		}
+
+		l.Error(err, "Cannot get PostgreSQL user", "user", u.Spec)
+		r.EventRecorder.Eventf(c, models.Warning, models.CreationFailed,
+			"Cannot get PostgreSQL user. user reference: %v", uRef)
+		return err
+	}
+
+	secret, err := v1beta1.GetDefaultPgUserSecret(context.TODO(), c.Name, c.Namespace, r.Client)
```

Comment:
Why you dont use ctx that we have in func signature?

Reply:
fixed